src/context/box-repository.ts

```diff
@@ -158,4 +158,31 @@ export class BoxRepository extends BaseRepository<BoxEntity> {
 
     return globalIndex;
   }
+
+  public async getAddressesBoxCount(options: {
+    where: { addresses: string[]; maxHeight?: number };
+  }) {
+    let baseQuery = this.repository
+      .createQueryBuilder("box")
+      .select("COUNT(*)", "boxesCount")
+      .addSelect("box.address", "address")
+      .where("box.address IN (:...addresses)")
+      .groupBy("box.address")
+      .setParameters(
+        removeUndefined({
+          addresses: options.where.addresses,
+          height: options.where.maxHeight
+        })
+      );
+    if (options.where.maxHeight) {
+      baseQuery = baseQuery
+        .innerJoin(
+          TransactionEntity,
+          "tx",
+          "tx.transactionId = box.transaction and tx.inclusionHeight <= :height"
+        )
```

Comment:
Hmm, I don't think this join is necessary

Reply:
I've deleted it.
The thing that caused me to make this mistake was that 2 of my boxes' creationHeight were equal to 0 in the database and it was odd.
I tough by adding this join I can prevent our app from returning such boxes.

Reply:
Oh, I see, nice catch! Actually that's my fault, I'm using `creationHeight` when we should be using `settlementHeight` in place of it.

`settlementHeight` is the height where the box is included in the chain, `creationHeight` should be defined on transaction building time, so it can be any value less than the actual pre-header height.

So, please replace `box.creationHeight` for `box.settlementHeight`. I'll push a fix for transactions and balance filtering.

Reply:
Aha, didn't know that about database schema!
Fixed and pushed.